scratchml/models/svc.py

```diff
@@ -0,0 +1,202 @@
+from scratchml.utils import split_data_into_batches
+from ..metrics import accuracy, recall, precision, f1_score, confusion_matrix
+from scratchml.regularizations import l2
+from scratchml.scalers import StandardScaler
+import numpy as np
+
+
+class SVC:
+    """
+    Support Vector Classifier (SVC) with options for linear, polynomial, and RBF kernels.
+    """
+
+    _valid_metrics = ["accuracy", "precision", "recall", "f1_score", "confusion_matrix"]
+
+    def __init__(
+        self,
+        C=0.35,
+        alpha=0.01,
+        kernel="linear",
+        degree=3,
+        max_iter=1500,
+        tol=1e-4,
+        learning_rate=1e-5,
+        decay=0.999,
+        batch_size=64,
+        early_stopping=True,
+        adaptive_lr=False,
+    ):
+        if C <= 0:
```

Comment:
Please, delete this if condition. You already do this verification in the `_validate_parameters` function.